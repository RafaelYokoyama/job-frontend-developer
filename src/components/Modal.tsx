import React from 'react';
import { ModalProps } from 'types/modal';
import Button from './Button';

const Modal = ({
  modalOpen,
  selectedVideo,
  closeModal,
  children
}: ModalProps) => {
  if (!modalOpen) return null;

  return (
    <div className="fixed z-10 inset-0 overflow-y-auto">
      <div className="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0">
        <div
          className="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full"
          role="dialog"
          aria-modal="true"
          aria-labelledby="modal-headline"
        >
          <div className="bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4">
            <iframe
              className="h-[200px] sm:h-[400px] w-full"
              src={`https://www.youtube.com/embed/${selectedVideo}`}
              title="YouTube video player"
              allowFullScreen
            ></iframe>
          </div>
          {children}
          <div className="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse">
            <Button
              onClick={closeModal}
              type="button"
              className="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm"
              text="Close"
            />
          </div>
        </div>
      </div>
    </div>
  );
};

export default Modal;
