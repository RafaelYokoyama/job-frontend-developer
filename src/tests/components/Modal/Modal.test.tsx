import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Modal from 'components/Modal';

const mockModalProps = {
  modalOpen: true,
  selectedVideo: 'abcd1234',
  closeModal: jest.fn(),
};

describe('Modal Component', () => {
  it('checks a snapshot', () => {
    const { container } = render(<Modal {...mockModalProps} />);
    expect(container).toMatchSnapshot();
  });
  it('should render modal with provided video', () => {
    const { getByTitle } = render(<Modal {...mockModalProps} />);
    const iframeElement = getByTitle('YouTube video player');
    expect(iframeElement).toBeInTheDocument();
    expect(iframeElement).toHaveAttribute(
      'src',
      'https://www.youtube.com/embed/abcd1234'
    );
  });

  it('should call closeModal function when close button is clicked', () => {
    const { getByText } = render(<Modal {...mockModalProps} />);
    const closeButton = getByText('Close');
    fireEvent.click(closeButton);
    expect(mockModalProps.closeModal).toHaveBeenCalled();
  });

  it('should not render modal when modalOpen is false', () => {
    const { queryByTitle } = render(
      <Modal {...mockModalProps} modalOpen={false} />
    );
    const iframeElement = queryByTitle('YouTube video player');
    expect(iframeElement).not.toBeInTheDocument();
  });

  it('renders children props correctly', () => {
    const mockChildElement = (
      <div data-testid="mock-child">Mock Child Element</div>
    );
    const { getByTestId } = render(
      <Modal {...mockModalProps}>{mockChildElement}</Modal>
    );
    const childElement = getByTestId('mock-child');
    expect(childElement).toBeInTheDocument();
    expect(childElement).toHaveTextContent('Mock Child Element');
  });
});
