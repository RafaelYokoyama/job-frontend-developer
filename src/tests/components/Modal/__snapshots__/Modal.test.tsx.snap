// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component checks a snapshot 1`] = `
<div>
  <div
    class="fixed z-10 inset-0 overflow-y-auto"
  >
    <div
      class="flex items-end justify-center min-h-screen pt-4 px-4 pb-20 text-center sm:block sm:p-0"
    >
      <div
        aria-labelledby="modal-headline"
        aria-modal="true"
        class="inline-block align-bottom bg-white rounded-lg text-left overflow-hidden shadow-xl transform transition-all sm:my-8 sm:align-middle sm:max-w-lg sm:w-full"
        role="dialog"
      >
        <div
          class="bg-white px-4 pt-5 pb-4 sm:p-6 sm:pb-4"
        >
          <iframe
            allowfullscreen=""
            class="h-[200px] sm:h-[400px] w-full"
            src="https://www.youtube.com/embed/abcd1234"
            title="YouTube video player"
          />
        </div>
        <div
          class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse"
        >
          <button
            class=" w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm"
            type="button"
          >
            <div
              class="flex items-center"
            >
              Close
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
